packages/rum-core/src/browser/domMutationObservable.ts

```diff
@@ -1,8 +1,62 @@
-import { monitor } from '@datadog/browser-core'
-import { LifeCycle, LifeCycleEventType } from '../domain/lifeCycle'
+import { monitor, Subscription } from '@datadog/browser-core'
 
-type MutationObserverConstructor = new (callback: MutationCallback) => MutationObserver
+export interface DOMMutationObservable {
+  subscribe(callback: () => void): Subscription
+}
+
+export function createDOMMutationObservable(): DOMMutationObservable {
+  let callbacks: Array<() => void> = []
+  const MutationObserver = getMutationObserverConstructor()
+  const observer = MutationObserver ? new MutationObserver(monitor(notify)) : undefined
+  let isDOMObserved = false
+
+  function notify() {
+    callbacks.forEach((callback) => callback())
+  }
```

Comment:
Could we use an observable internally rather than reimplementing the observable logic here?

Reply:
I also thought about it, but I have to hook into the observable logic to start and stop the MutationObserver which is currently not straightforward with the Observable is.

Reply:
what about something like https://github.com/DataDog/browser-sdk/commit/f9afcc2b4ab5613deac42f373999d83da89e22d5?